src/ORM/DataQuery.php

```diff
@@ -1555,4 +1573,42 @@ private function withCorrectDatabase(callable $callback): mixed
         }
         return $callback();
     }
+
+    private function filterOrExcludeByQuery(DataQuery $subQuery, bool $isExclude, string $fieldToFilterBy, string $fieldInOtherList): static
+    {
+        if ($fieldInOtherList === '' || $fieldToFilterBy === '') {
+            throw new InvalidArgumentException('$fieldToFilterBy and $fieldInOtherList must not be empty strings');
+        }
+        $subSelect = $subQuery->getFinalisedQuery();
+        // Select only the relevant field, and ID. The ID is used below to validate the join.
+        $subSelect->setSelect([]);
+        $subSelect->selectField($subQuery->expressionForField($fieldInOtherList), $fieldInOtherList);
+        $subSelect->selectField($subQuery->expressionForField('ID'), 'ID');
```

Comment:
```suggestion
        $subSelect->selectField($subQuery->expressionForField($fieldInOtherList));
        $subSelect->selectField($subQuery->expressionForField('ID'));
```
We don't need the alias arg?

Reply:
Not sure if we do or not, I'd prefer to keep it since it was there in the implementation of `subtract()` which this is based off.
Either:
1. it was there for an undocumented purpose, e.g. to solve some edge case (in which case we need it)
2. it was there for no specific purpose but does no harm being there

i.e. either we need it for reasons we don't know, or it does no harm so there's no specific reason to remove it.